service-tests/github.js

```diff
@@ -0,0 +1,60 @@
+'use strict';
+
+const Joi = require('joi');
+const ServiceTester = require('./runner/service-tester');
+
+const t = new ServiceTester({ id: 'github', title: 'GitHub' });
+module.exports = t;
+
+t.create('File size')
+  .get('/size/webcaetano/craft/build/craft.min.js.json')
+  .expectJSONTypes(Joi.object().keys({
+    name: Joi.equal('size'),
+    value: Joi.string().regex(/^[0-9]*[.]?[0-9]+\s(B|kB|MB|GB|TB|PB|EB|ZB|YB)$/),
+  }));
+
+t.create('File size 404')
+  .get('/size/webcaetano/craft/build/does-not-exist.min.js.json')
+  .expectJSONTypes(Joi.object().keys({
+    name: Joi.equal('size'),
+    value: Joi.string().regex(/^repo or file not found$/),
+  }));
+
+t.create('File size unknown')
+  .get('/size/webcaetano/craft/build.json')
+  .expectJSONTypes(Joi.object().keys({
+    name: Joi.equal('size'),
+    value: Joi.string().regex(/^unknown file$/),
+  }));
```

Comment:
Sorry, I'm still not being clear enough.

Could you set `badgeData.text[1] = 'not a regular file'` so the user sees that message on the badge?